packages/flutter/lib/src/material/material_state.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show Color;
+
+/// Interactive states that some of the Material widgets can take on when
+/// receiving input from the user.
+///
+/// States are defined by https://material.io/design/interaction/states.html#usage.
+///
+/// Some Material widgets track their current state in a `Set<MaterialState>`.
+///
+/// See also:
+///  * [MaterialStateColor], a color that has a `resolve` method that can
+///    return a different color depending on the state of the widget that it
+///    is used in.
+enum MaterialState {
```

Comment:
In a future version we'll probably want to add "indeterminate" for things like tri-state checkboxes and toggle buttons.

Reply:
Good catch, leaving this out for now since it's not one of Material's main "States", but we may want to add this later.